src/qz/common/TrayManager.java

```diff
@@ -182,14 +188,34 @@ public TrayManager(boolean isHeadless) {
                                 }
                             });
                         }
-                    } catch(InterruptedException ignore) {}
+                    }
+                    catch(InterruptedException ignore) {}
                 }
             }).start();
         }
 
         if (tray != null) {
             addMenuItems();
         }
+
+        // Initialize idle actions
+        idleTimers = new ArrayList<>();
+
+        // Slow to find printers the first time if a lot of printers are installed
+        performIfIdle((int)TimeUnit.SECONDS.toMillis(10), evt -> {
+            log.debug("IDLE: Performing first run of find printers");
+            PrintServiceMatcher.getNativePrinterList(true);
```

Comment:
This is only half the battle... what takes a bit is the registry/shell stuff that gets called per-printer.

Reply:
Added proactive fetching of all the driver details per 3f133fa